spv/sync.go

```diff
@@ -78,7 +78,7 @@ type Syncer struct {
 
 	// Mempool for non-wallet-relevant transactions.
 	mempool     sync.Map // k=chainhash.Hash v=*wire.MsgTx
-	mempoolAdds chan *chainhash.Hash
+	mempoolAdds chan chainhash.Hash
```

Comment:
Txs were not being removed from the mempool map because they are keyed by `chainhash.Hash` but removal attempts used `*chainhash.Hash`.